torch/csrc/jit/python/python_sugared_value.cpp

```diff
@@ -438,8 +438,14 @@ std::shared_ptr<SugaredValue> SugaredEnumClass::attr(
 }
 
 SugaredValuePtr SugaredEnumClass::iter(const SourceRange& loc, Function& m) {
-  // TODO(gmagogsfm): Implement getting iterator.
-  TORCH_INTERNAL_ASSERT(false);
+  std::vector<Value*> enum_pure_values;
+  enum_pure_values.reserve(enum_values_.size());
+  for (const auto& name_sv : enum_values_) {
+    enum_pure_values.push_back(name_sv.second->asValue(loc, m));
+  }
+  Node* enum_pure_values_list = m.graph()->insertNode(
+      m.graph()->createList(enum_type_, enum_pure_values));
```

Comment:
I wonder if it would be better to create this list upfront as a constant like we do with the enum values themselves and retrieve it here. If it ends up not being used, it'll be optimized out?

Do you know if there is already an optimization that recognizes that this list construction involves only graph constants and does not perform it at runtime?

Reply:
My worry is that this creates a `prim::ListConstruct` node that creates the list at runtime through a corresponding `LIST_CONSTRUCT` interpreter op. But we know at compile time how long this list is and what its contents are, so we can avoid constructing the list at run time.

Reply:
I am probably missing something. I thought that would get constant folded and become a list constant?

Reply:
If that's the case, that's fine too. I think you can probably print the `graph` attribute in your test to find out for sure?

Reply:
Great catch! runCleanUpPasses actually doesn't run constant folding on mutable types (list). Just changed implementation make sure a list constant is generated.

Reply:
Wait, why did you change it? Did you find out for sure that constant folding does not occur?